docs/src/pages/customization/components/DynamicThemeNesting.tsx

```diff
@@ -27,22 +39,7 @@ export default function DynamicThemeNesting() {
         }
         label="Blue"
       />
-      <ThemeProvider
-        theme={
-          color === 'blue'
-            ? {
-                ...defaultTheme,
-                palette: {
-                  ...defaultTheme.palette,
-                  secondary: {
```

Comment:
The merge is not deep which is now explicit with types. Fixes
![Screenshot from 2019-11-22 13-21-55](https://user-images.githubusercontent.com/12292047/69425766-d6663a00-0d2b-11ea-93b3-f4d9dfb91984.png)

Hovered:
Old 
![Screenshot from 2019-11-22 13-23-46](https://user-images.githubusercontent.com/12292047/69425702-bafb2f00-0d2b-11ea-9f8e-5029fa480f9f.png)
New
![Screenshot from 2019-11-22 13-24-09](https://user-images.githubusercontent.com/12292047/69425711-bf274c80-0d2b-11ea-882a-b46fdc0fcb23.png)